src/sentry/api/bases/monitor.py

```diff
@@ -57,23 +60,64 @@ class MonitorEndpoint(Endpoint):
     def respond_invalid() -> Response:
         return Response(status=status.HTTP_400_BAD_REQUEST, data={"details": "Invalid monitor"})
 
-    def convert_args(self, request: Request, monitor_id: str, *args, **kwargs):
-        try:
-            UUID(monitor_id)
-        except ValueError:
-            raise ParameterValidationError("Invalid monitor UUID")
+    def convert_args(
+        self,
+        request: Request,
+        monitor_id: str,
+        organization_slug: str | None = None,
+        *args,
+        **kwargs,
+    ):
+        organization = None
+        monitor = None
+
+        # The only monitor endpoints that do not have the org slug in their
+        # parameters are the GUID-style checkin endpoints
+        if organization_slug:
+            try:
+                organization = Organization.objects.get_from_cache(slug=organization_slug)
+            except Organization.DoesNotExist:
+                pass
+
+        # Try lookup by slug first. This requires organization context since
+        # slugs are uniuqe only to the organization
+        if organization:
+            try:
+                monitor = Monitor.objects.get(organization_id=organization.id, slug=monitor_id)
+            except Monitor.DoesNotExist:
+                pass
```

Comment:
stick this under the clause on line 76

Reply:
What do you mean?

Reply:
`organization` can only have a value if `organization_slug` is set. so you can move the `monitor` check into one `try...except` that catches both `Organization.DoesNotExist` and `Monitor.DoesNotExist`. sorry for lack of specificity my bad